### PR TITLE
Fix error handling in sharepoint reader

### PR DESCRIPTION
**Pull request by Javtor**
# Description

For some errors we weren't returning the appropriate error description according to the failed request to Microsoft Graph API

Fixes https://github.com/run-llama/llama_index/issues/15918

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- [x] Manually tested locally

### CODE REVIEW

File: llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py [hunk 1]
```diff
@@ -788,10 +800,8 @@ def load_resource(self, resource_id: str, **kwargs) -> List[Document]:
 
             item = self._get_item_from_path(path)
 
-            input_file_dir = path.parent
-
             with tempfile.TemporaryDirectory() as temp_dir:
-                metadata = self._download_file(item, temp_dir, input_file_dir)
```
Comment (Javtor) [inline llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py:794]: this is another bugfix - we were passing an extra unexpected param to `_download_file`

File: llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py [hunk 2]
```diff
@@ -116,12 +116,17 @@ def _get_access_token(self) -> str:
...
+        json_response = response.json()
+
+        if response.status_code == 200 and "access_token" in json_response:
+            return json_response["access_token"]
 
         else:
-            logger.error(response.json()["error"])
-            raise ValueError(response.json()["error_description"])
+            error_message = json_response.get("error_description") or json_response.get(
+                "error"
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py:126]: Should we default to something a little more descriptive? "Unknown error retrieving access code" maybe?

File: llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py [hunk 3]
```diff
@@ -209,26 +215,27 @@ def _get_drive_id(self) -> str:
...
+                return json_response["value"][0]["id"]
             else:
                 raise ValueError(
                     "Error occurred while fetching the drives for the sharepoint site."
                 )
         else:
-            logger.error(response.json()["error"])
-            raise ValueError(response.json()["error_description"])
+            error_message = json_response.get("error_description") or json_response.get(
+                "error"
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py:235]: Similar comment

**Comment by logan-markewich** [comment]
@Javtor Can you add some better error messages here?

This will also fix https://github.com/run-llama/llama_index/issues/15918

**Comment by Javtor** [comment]
> @Javtor Can you add some better error messages here?
> 
> This will also fix #15918

made the errors more descriptive

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**